track.go

```diff
@@ -4,41 +4,70 @@ import "time"
 
 var _ Message = (*Track)(nil)
 
-// This type represents object sent in a track call as described in
-// https://segment.com/docs/libraries/http/#track
 type Track struct {
 	// This field is exported for serialization purposes and shouldn't be set by
 	// the application, its value is always overwritten by the library.
-	Type string `json:"type,omitempty"`
-
-	MessageId    string       `json:"messageId,omitempty"`
-	AnonymousId  string       `json:"anonymousId,omitempty"`
-	UserId       string       `json:"userId,omitempty"`
-	Event        string       `json:"event"`
-	Timestamp    time.Time    `json:"timestamp,omitempty"`
-	Context      *Context     `json:"context,omitempty"`
-	Properties   Properties   `json:"properties,omitempty"`
-	Integrations Integrations `json:"integrations,omitempty"`
+	Type           int    `json:"t"`
+	PayloadVersion int    `json:"v"`
+	Project        string `json:"p"`
+
+	Timestamp    time.Time `json:"ts"`
+	MessageId    string    `json:"mid"`
+	InstanceId   string    `json:"iid"`
+	DeploymentId string    `json:"did"`
+
+	OsName         string `json:"osn,omitempty"`
+	OsArchitecture string `json:"osa,omitempty"`
+	CPU            int    `json:"cpu,omitempty"`
+
+	// Alloc is bytes of allocated heap objects.
+	Alloc uint64 `json:"alloc"`
```

Comment:
Will these be sent always? If not, please make them `omitempty`. Please check that `omitempty`actually removes them from the JSON output, if not, they have to be pointers.

Reply:
These are sent always. CPU is a new addition and probably it could be also without ommitempty.